build-plugin/action.yml

```diff
@@ -81,6 +81,29 @@ runs:
         printf "plugin-id=%s\n" "${{ steps.package-plugin.outputs.plugin-id }}"
         printf "plugin-version=%s\n" "${{ steps.package-plugin.outputs.plugin-version }}"
 
+    - name: Checkout code
+      if: ${{ env.USE_CHANGELOG_GENERATOR }}
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: main
```

Comment:
The default branch is often called `main` but that might not be the case. I think we should make this a bit more flexible:
- If we remove the `ref` argument, it should default to the default branch (see https://github.com/actions/checkout?tab=readme-ov-file#usage)
- Otherwise, we can have an input with the default branch name, which will default to `main`. This is probably better so we can also pass it to the "Commit updated changelog" step down the workflow

Reply:
Isn't the code already checked out from the call site? (As per [readme's workflow example](https://github.com/grafana/plugin-actions/blob/main/build-plugin/README.md#workflow-example)).